pkg/virt-controller/watch/vmi.go

```diff
@@ -435,6 +455,10 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 		if allDeleted {
 			log.Log.V(3).Object(vmi).Infof("All pods have been deleted, removing finalizer")
 			controller.RemoveFinalizer(vmiCopy, virtv1.VirtualMachineInstanceFinalizer)
+			if vmiCopy.Labels != nil {
+				delete(vmiCopy.Labels, virtv1.OutdatedLauncherImageLabel)
```

Comment:
I actually can't find where is this being used... :/

If it's not being used, why do we need to worry about deleting it, even if it is, seems that the vmi is deleted at this.
I think I missing something.

Reply:
The label only exists for convenience. It serves as a way for admins to identify and list all VMIs which are out of date using a label selector. I wouldn't care about this if we had a way to perform field selection for VMIs, but we don't :( 

So you're right that our controllers don't perform any action based on this label. Instead our controllers look at the VMI status's container image to determine if the VMI is out of date. But, with the label, a field engineer could easily identify out of 1000s of VMs which ones are not successfully updating. 

I originally had the label trigger our controllers as well, making the label serve both purposes. However, roman talked me out of that because the labels are mutable by end users where our VMI status is not.

> If it's not being used, why do we need to worry about deleting it, even if it is, seems that the vmi is deleted at this.
I think I missing something.

i remove the label when the VM is in a finalized state. That doesn't necessarily mean the VMI is deleted.  It could be that the VMI exited on it's own and the VMI object will still stick around for a bit. Removing the label ensures anyone filtering by this label will only see active VMIs.

Reply:
Thanks for the explanation!

> i remove the label when the VM is in a finalized state. That doesn't necessarily mean the VMI is deleted. It could be that the VMI exited on it's own and the VMI object will still stick around for a bit. Removing the label ensures anyone filtering by this label will only see active VMIs.

Yea. It makes sense to me now. I didn't understand that there will be something dependent on this label.

Reply:
Hey @davidvossel!
Can you please update the PR description to align with what's explained here? IIUC, the following is now incorrect:

> this label is used by the automated workload update controller to detect outdated VMIs. It's also helpful for users who may want to create their own logic to handle outdated VMIs external of the automation in this PR.